### ISSUE TITLE
Solved: Move logs and cache dires of the Portable App to a different location?

### ISSUE THREAD

**Issue by maelcum** [canton7/SyncTrayzor#294]
Hello.

tl;dr
Is there any way to move the SyncTrayzorPortable-x64/logs-directory to some other location (like %AppData%/SyncTrayzor/log)?

Long version:

SyncTrayzor.exe version 1.1.11.0
Syncthing.exe version 0.13.10
Windows 10 version Prio N 1511 build 10586.420

I've read about the difference between the setup and portable versions of SyncTrayzor and decided for the portable (x64) version. It has been installed in directory that is synced with Microsoft OneDrive. In SyncTrayzors settings the <SyncthingCustomHomePath> was changed to "C:\Users\<me>\AppData\Local\Syncthing".

That way I hope(d) to retain the binary and it's settings while having all temporary data on the local machine, way outside whatever OneDrive is syncing.

Then I learned that SyncTrayzor is storing some temporary data in 
SyncTrayzorPortable-x64/data/cef/cache/... 
SyncTrayzorPortable-x64/logs/... as well as in 
SyncTrayzorPortable-x64/debug.log. 
Now these files change quite frequently, thus triggering OneDrive again and again *1. Is there a way (maybe start parameters or registry entries to make) that would allow me to shift as much of that data to some other location as well?

Thanks for programming and sharing SyncTrayzor in the first place. :)
Thanks for reading this.
Please excuse all späling errors and wöird sentence structure.
Have a nice day.

Wolf

*1 Why this should be a problem? OneDrive is a sync engine and should cope with that, shouldn't it? Well, yes is *should*. OD has this peculiar behaviour of sinking its teeth firmly into a file that keeps changing a lot, by that ignoring all the other files that accumulate over time and would like to be synced as well. This becomes a minor annoyance as it forces me to stop SyncTrayzor now and then, restart OneDrive, wait for OD to do its thing and restart SyncTrayzor, at which point OD is completely going apesh*t again... There is no automatism anymore and I can never be sure the oneline version of OD is synchronized with the local version.]

**Comment by canton7** [comment]
Have a look at the Path configuration section of SyncTrayzor.exe.config. You can use env vars in there if you want. 

Note that upgrades will overwrite this file (it's intended to be edited only be system administrators who are creating their own deployment packages for their machines). 

Also be aware that syncing Syncthing's home directory (which is also stored in data/) is a very bad idea: it's device id is stored in there, as well as its index, neither of which should be the same on any two of your machines.

**Comment by canton7** [comment]
It might be easier to use the installed version of SyncTrayzor, and edit SyncTrayzor.exe.config to locate it's configuration file in a OD folder. The auto-upgrade mechanism is pretty smooth, so I wouldn't worry about synchronising binaries. In fact, there aren't really my options on the configuration file which you should need to change after an initial setup, so I'm not sure i'd even synchronise that.

**Comment by maelcum** [comment]
Thanks for the quick response.
Syncthings home dir has already been moved and does not show up in .../data. 
The original post should have said: 

> In SyncTrayzors settings < config.xml > the < SyncthingCustomHomePath > was changed to "C:\Users<me>\AppData\Local\Syncthing".

Unfortunately, the < > were stripped with their contents. I'm sorry.

**Comment by canton7** [comment]
My bad, I'm in the middle of something else, and missed that.

**Comment by maelcum** [comment]
> It might be easier to use the installed version of SyncTrayzor...

I agree. My initial understanding was that the portable version would be self-contained whereas the setup version would have dependencies and such. Wrong assumption.

Since the portable version is already installed I experimented with the SyncTrayzor.exe.config as advised and it worked like a charm. Oh if only everything in life would be that easy. THANK. YOU (esp. for thinking about this configuration options in the first place.)

Just for documentations' sake, what has been changed:
```
<LogFilePath>%LocalAppData%\SyncTrayzor\logs</LogFilePath >
<CefCachePath>%LocalAppData%\SyncTrayzor\data\cef\cache</CefCachePath>
<SyncthingCustomHomePath>%LocalAppData%\Syncthing</SyncthingCustomHomePath>
```
and some other options that could've been configured in the GUI as well.
SyncTrayzor creates all necessary dirs and starts working without a hiccup.
The only file left in the binaries-dir is the debug.log, with which OD can hopefully live. It seems not to change too often.

**Comment by canton7** [comment]
> I agree. My initial understanding was that the portable version would be self-contained whereas the setup version would have dependencies and such. Wrong assumption.

Yeah, they're both self-contained in that the installer / portable zip contains everything you need (apart from the .NET framework, which only the installer will install). The intention with the portable version is that you can installer it to e.g. a USB stick, so everything - log files, config files, etc - is stored in the one directory.
